src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java

```diff
@@ -234,8 +242,8 @@ public ProgrammingExercise createProgrammingExercise(ProgrammingExercise program
         // make sure that plagiarism detection config does not use existing id
         Optional.ofNullable(programmingExercise.getPlagiarismDetectionConfig()).ifPresent(it -> it.setId(null));
 
-        // for LocalCI and Aeolus, we store the build plan definition in the database as a windfile
-        if (aeolusTemplateService.isPresent() && programmingExercise.getBuildPlanConfiguration() == null) {
+        // for LocalCI and Aeolus, we store the build plan definition in the database as a windfile, we don't do that for Jenkins
```

Comment:
Why don’t we do that for Jenkins? Some short explanation in the comment here would be helpful for others.

Reply:
For Jenkins and Bamboo we only store the windfile if needed, in the case of Localci we always need it, I'll clarify this in a followup along with the other improvements.